alert-engine/src/main/scala/io/phdata/pulse/alertengine/AlertEngineImpl.scala

```diff
@@ -75,13 +75,15 @@ class AlertEngineImpl(solrServer: CloudSolrServer, notificatonServices: Notifica
         query.set("collection", alias)
         val results   = solrServer.query(query).getResults.asScala
         val threshold = alertRule.resultThreshold.getOrElse(0)
-
         if (threshold == -1 && results.isEmpty) {
-          logger.info(s"Alert triggered for $applicationName on alert $alertRule")
-          Some(TriggeredAlert(alertRule, applicationName, results))
+          logger.info(
+            s"Alert triggered for $applicationName on alert $alertRule at no results found condition")
+          Some(TriggeredAlert(alertRule, applicationName, results, 0))
         } else if (results.lengthCompare(threshold) > 0) {
           logger.info(s"Alert triggered for $applicationName on alert $alertRule")
-          Some(TriggeredAlert(alertRule, applicationName, results))
+          query.setRows(0)
```

Comment:
This line can be deleted

Reply:
Do you mean to delete the "query.setRows(0)"? this line will return only numbers of result, not with all the real documents.